<launch>
  <arg name="joy_type" default="DS4" doc="options: DS4, G29"/>

  <arg name="input_joy" default="/joy" />
  <arg name="input_twist" default="/localization/twist" />

  <arg name="output_control_command" default="/remote/joy/control_cmd" />
  <arg name="output_raw_control_command" default="/remote/raw_control_cmd" />
  <arg name="output_shift" default="/remote/shift_cmd" />
  <arg name="output_turn_signal" default="/remote/turn_signal_cmd" />
  <arg name="output_gate_mode" default="/remote/gate_mode_cmd" />
  <arg name="output_emergency_stop" default="/remote/emergency_stop" />
  <arg name="output_autoware_engage" default="/autoware/engage" />
  <arg name="output_vehicle_engage" default="/vehicle/engage" />

  <arg name="config_file" default="$(find-pkg-share autoware_joy_controller)/config/autoware_joy_controller.yaml" />

  <node pkg="autoware_joy_controller" exec="autoware_joy_controller" name="autoware_joy_controller" output="screen">
    <remap from="input/joy" to="$(var input_joy)"/>
    <remap from="input/twist" to="$(var input_twist)"/>

    <remap from="output/control_command" to="$(var output_control_command)"/>
    <remap from="output/raw_control_command" to="$(var output_raw_control_command)"/>
    <remap from="output/shift" to="$(var output_shift)"/>
    <remap from="output/turn_signal" to="$(var output_turn_signal)"/>
    <remap from="output/gate_mode" to="$(var output_gate_mode)"/>
    <remap from="output/emergency_stop" to="$(var output_emergency_stop)"/>
    <remap from="output/vehicle_engage" to="$(var output_vehicle_engage)"/>
    <remap from="output/autoware_engage" to="$(var output_autoware_engage)"/>
    <remap from="output/vehicle_engage" to="$(var output_vehicle_engage)"/>

    <!-- tmp -->
    <remap from="output/vehicle_cmd" to="/remote/vehicle_cmd"/>
    <remap from="output/raw_vehicle_cmd" to="/remote/raw_vehicle_cmd"/>

    <remap from="service/clear_emergency_stop" to="/control/vehicle_cmd_gate/clear_external_emergency_stop"/>

    <param from="$(var config_file)" />
    <param name="joy_type" value="$(var joy_type)" />
  </node>

  <node pkg="joy" exec="joy_node" name="joy_node" output="screen">
    <param name="autorepeat_rate" value="1.0" />
    <param name="deadzone" value="0.0" />
  </node>
</launch>
